### PR TITLE
DROOLS-3703: Disable Enumeration and Range constraint for Any DataType

### PR DESCRIPTION
**Pull request by jomarko**
For more details see: https://issues.redhat.com/browse/DROOLS-3703

Effect of these changes can be seen in https://issues.redhat.com/secure/attachment/12783227/DROOLS-3703-fix.webm

### CODE REVIEW

**Comment by tiagobento** [comment]
@jomarko What's the status of this PR? Still draft?

**Comment by jomarko** [comment]
@tiagobento it is a draft yes, however my actual code changes are finished. I am just waiting for some approval, that the solution I prepared is acceptable approach and then I will finish tests for this PR and convert to the regular PR. @yesamer would you mind to have a look?

@tiagobento does it make sense a little bit please?

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalView.java [hunk 1]
```diff
@@ -82,19 +86,20 @@ public class DataTypeConstraintModalView implements DataTypeConstraintModal.View
...
+            final HTMLDivElement body,
+            final HTMLDivElement footer,
+            final HTMLDivElement componentContainer,
+            final HTMLButtonElement okButton,
+            final HTMLButtonElement cancelButton,
+            final HTMLAnchorElement clearAllAnchor,
+            final @Named("span") HTMLElement type,
+            final HTMLDivElement selectConstraint,
+            final HTMLDivElement constraintWarningMessage,
+            final HTMLButtonElement closeConstraintWarningMessage) {
```
Comment (yesamer) [inline packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalView.java]: @jomarko Is this probably automatic formatting (not intended)?

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalTest.java [hunk 2]
```diff
@@ -206,7 +206,7 @@ public void testPrepareViewWhenConstraintValueIsBlank() {
         doReturn(type).when(modal).getConstraintValueType();
         doReturn(constraint).when(modal).getConstraintValue();
 
-        modal.prepareView();
+//        modal.prepareView();
```
Comment (yesamer) [inline packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalTest.java]: @jomarko What about these commented code?

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintTest.java [hunk 3]
```diff
@@ -165,7 +165,7 @@ public void testOpenModal() {
 
         dataTypeConstraint.openModal();
 
-        constraintModal.load(type, constraint, ENUMERATION);
+//        constraintModal.load(type, constraint, ENUMERATION);
```
Comment (yesamer) [inline packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintTest.java]: @jomarko What about this commented code?

**Review by yesamer — CHANGES_REQUESTED** [review]
@jomarko Thank you, your changes look fine to me. Just correct the commented code and we can merge it.

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalView.java (hunk 1 above)
Comment (jomarko) [reply packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalView.java]: yes, thank you, reverting

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalTest.java (hunk 2 above)
Comment (jomarko) [reply packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModalTest.java]: fixing

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModal.java [hunk 4]
```diff
@@ -106,17 +110,17 @@ void doSave(final String value) {
...
 
-    void load(final String type,
-              final String value,
-              final ConstraintType constraintType) {
+    void load(final DataTypeListItem dataTypeListItem) {
 
-        this.constraintValue = value;
-        this.constraintValueType = type;
+        final String constraintValue = dataTypeListItem.getDataType().getConstraint();
+        this.constraintValue = constraintValue;
```
Comment (yesamer) [inline packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModal.java]: Optional: you can avoid the locally scoped variable and use the field instead.

**Review by yesamer — APPROVED** [review]
@jomarko Nice changes, thank you! I just have an optional suggestion.

**Comment by yesamer** [comment]
@jomarko Please ping me to merge it.

File: packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModal.java (hunk 4 above)
Comment (jomarko) [reply packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/DataTypeConstraintModal.java]: thank you for spotting

**Comment by jomarko** [comment]
@yesamer please merge

**Pull request merged**